enhancements/disaster-recovery/automated-cluster-backups.md

```diff
@@ -0,0 +1,324 @@
+lost. It might make sense to document the requirement to periodically ship
+backup data off the cluster. Given that the current restore procedure
+requires at least one control plane node, though, a cluster is only
+recoverable from backup if at least one of the backup-containing control
+plane nodes is available.
+
+## Design Details
+
+### Open Questions
+
+- Under what circumstances should backup pods that do not terminate
+  successfully or do not terminate within a reasonable period of time be
+  recreated?
+
+- If the node targeted for backup does not have sufficient space, should
+  there be an attempt to take a backup on a different node? This question is
+  complicated by spreading backups across available nodes since backing up to
+  a different node may require removing a backup that is not the oldest.
+
+### Test Plan
+
+- Comprehensive unit testing of
+  - Changes in backup configuration
+  - Scheduling logic
+  - Backup retention
+  - Pre-backup check for disk space available
+
+- Update DR e2e testing to initiate one-time backups via the controller
+
+- Add new e2e golden-path testing of scheduled backups
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, or as something else. Initial proposal
+should keep this high-level with a focus on what signals will be looked at to
+determine graduation.
+
+Consider the following in developing the graduation criteria for this
+enhancement:
+
+- Maturity levels
+  - [`alpha`, `beta`, `stable` in upstream Kubernetes][maturity-levels]
+  - `Dev Preview`, `Tech Preview`, `GA` in OpenShift
+- [Deprecation policy][deprecation-policy]
+
+Clearly define what graduation means by either linking to the [API doc definition](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning),
+or by redefining what graduation means.
+
+In general, we try to use the same stages (alpha, beta, GA), regardless how the functionality is accessed.
+
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+**Examples**: These are generalized examples to consider, in addition
+to the aforementioned [maturity levels][maturity-levels].
+
+#### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+- Enumerate service level indicators (SLIs), expose SLIs as metrics
+- Write symptoms-based alerts for the component(s)
+
+#### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+- Backhaul SLI telemetry
+- Document SLOs for the component
+- Conduct load testing
```

Comment:
In this particular instance, should load testing move to the dev preview section?

Reply:
Also, the metrics and SLIs...should be considered during dev preview phase.

Reply:
I don't think this is a 'dev preview' type of feature. @Anandnatraj can confirm, but I believe the urgency of this feature suggests providing it GA in whatever release it is first included in.